src/api/Nhs.Appointments.Core/Sites/SiteService.cs

```diff
@@ -41,6 +41,8 @@ public class SiteService(
     IFeatureToggleHelper featureToggleHelper,
     IOptions<SiteServiceOptions> options) : ISiteService
 {
+    private static readonly SemaphoreSlim _cacheLock = new(1, 1);
```

Comment:
This only ensures thread safety because I've changed the service registration to register this class as a singleton. Even then, it only ensures thread safety in one instance of our http functions app. If we scale out we could still end up with race conditions. 

To do this properly, we'd need to implement an[ IDistributed cache](https://learn.microsoft.com/en-us/aspnet/core/performance/caching/distributed?view=aspnetcore-9.0). I propose that this is good enough for now and we should bring in a distributed cache as future work

Reply:
Yeah I'd had exactly the same thought, but I decided it would be better to do this when we introduce the IDistributed cache else we're kind of duplicating effort. 

Happy to do it as part of this work if you want, but think my preference is to defer. In this cache we know sites is an array and we know it's identifiable by an ID. If we created a generic caching interface we'd have to cater for the cases where these two things aren't true, which is just more work than I intended to do in this particular PR.